redis/src/main/scala/zio/redis/RespValue.scala

```diff
@@ -146,16 +150,24 @@ object RespValue {
               case state              => CollectingArray(rem, vals, state.feed)
             }
 
-          case ExpectingBulk => Done(bulkString(line))
-          case _             => Failed
+          case CollectingBulkString(rem, vals, index) =>
+            val allLineBytes = Chunk.fromArray(line.getBytes(StandardCharsets.UTF_8))
```

Comment:
I'm wondering if it's possible to postpone (or limit) the UTF8 encode & decode, as it will be quite expensive. :thinking:

Reply:
We can append bulk string lines using a stringbuilder and then, only at the end encoding as UTF bytes.